code-challenges.test.js

```diff
@@ -20,7 +20,38 @@ var colors1 = ["purple", "blue", "green", "yellow", "pink"]
 var colors2 = ["chartreuse", "indigo", "periwinkle", "ochre", "aquamarine", "saffron"]
 // Expected output example (can be a different order): ["saffron", "aquamarine", "periwinkle", "indigo", "ochre"]
 
-
+// a describe method that lists the name of the function OR naming of the particular test.
+describe("shuffleTheRest", () => {
+    let colors1 = ["purple", "blue", "green", "yellow", "pink"]
+    let colors2 = ["chartreuse", "indigo", "periwinkle", "ochre", "aquamarine", "saffron"]
+    // a test/it method, nested within the describe block, that in plain words, describes that the function does.
+    it("takes in an array, removes the first item from the array and shuffles the remaining content.", () => {
+
+        //an expect method, nested within the test block, calling on the shuffleTheRest() function, followed by the .toEqual() matcher that checks the expected output of the function return.
+        expect(shuffleTheRest(colors1)).toEqual(expect.arrayContaining(["yellow", "blue", "pink", "green"]))
+        expect(shuffleTheRest(colors2)).toEqual(expect.arrayContaining(["saffron", "aquamarine", "periwinkle", "indigo", "ochre"]))
+    })
+})
+// -- Got the red --  
+
+// function shuffleTheRest(array) {
+//     // call shift method (remove) the first element of the array:
+//     array.shift();
+//     // randomizer and sort the array
+//     const shuffledArray = array.sort((a, b) => 0.5 - Math.random());
+//     //return the array
+//     return shuffledArray
+// }
+// console.log(shuffleTheRest(colors1));
+
+// -- Got the green -- 
+
+// -- Refactor --  
+
+function shuffleTheRest(array) {
+    array.shift();
+    return shuffledArray = array.sort((a, b) => 0.5 - Math.random()); 
+}
```

Comment:
I like the refactor. In this case, I don't think you need to pass any arguments to the sort method.